demo/flake.nix

```diff
@@ -1,54 +1,101 @@
+
+    # Possible values for compilationType: "local-native", "remote-native", or "cross".
+    compilationType = "cross"; # Choose the compilation type here.
+
+    localSystem = "x86_64-linux";
+    targetSystem = "aarch64-linux";
+
+    # Kernel packages based on compilationType (local native, remote native, or cross-compilation)
+    pkgsKernel =
+      if compilationType == "cross"
+      then
+        import nixpkgs {
+          inherit localSystem;
+          crossSystem = targetSystem;
+        }
+      else if compilationType == "local-native"
+      then import nixpkgs {system = targetSystem;}
+      else
+        # For remote-native compilation
+        import nixpkgs {system = targetSystem;};
+
+    # Define bootloader based on bootType (UEFI or U-Boot)
+    bootloaderModule =
+      if bootType == "uefi"
+      then
+        # UEFI configuration with bootloader settings
+        {
+           # grub bootloader configured for UEFI
+           boot = {
+             # growPartition = true;  # If partition resizing is necessary
+             kernelParams = ["console=ttyS0"];  # If you need serial console access
+             # loader.timeout = lib.mkDefault 0;  # Optional, to skip GRUB menu
+             initrd.availableKernelModules = ["uas"];  # If specific kernel modules are required
+             loader.grub = {
+               enable = true;
+               device = "nodev";
+               efiSupport = true;
+               efiInstallAsRemovable = true;
+             };
+           };
+        }
+      else
+        # U-Boot configuration using sd-image
+        nixos-rk3588.nixosModules.orangepi5.sd-image;
   in {
     colmena = {
       meta = {
-        nixpkgs = import nixpkgs {inherit system;};
-        rk3588 = {
-          inherit nixpkgs pkgsKernel;
+        # Use the correct package set for the system.
+        nixpkgs = import nixpkgs {system = localSystem;};
+        specialArgs = {
+          rk3588 = {
+            inherit nixpkgs pkgsKernel;
+          };
+          inherit nixpkgs;
         };
       };
 
+      ## TODO: possibly need to change "opi5" to "orangepi5" or "orangepi5plus", the local host name if building locally
       opi5 = {
-        deployment.targetHost = "192.168.5.42";
-        deployment.targetUser = "root";
-        # Allow local deployment with `colmena apply-local`
-        # deployment.allowLocalDeployment = true;
+        deployment.targetHost =
+          if compilationType != "cross"
+          then "192.168.5.42"
+          else null;
+        deployment.targetUser =
+          if compilationType != "cross"
+          then "root"
+          else null;
+
+        # Allow local deployment only if building locally or if remote-native
+        deployment.allowLocalDeployment = compilationType == "local-native";
 
         imports = [
-          # import the rk3588 module, which contains the configuration for bootloader/kernel/firmware
-          boardModule.core
```

Comment:
The custom kernel is defined in `nixos-rk3588.nixosModules.orangepi5.core`, and removing it here would cause the resulting NixOS system to use the mainline kernel, which seems like it could be problematic.